Assets/MixedRealityToolkit.Extensions/Webrtc/Scripts/WebrtcPeerVideo.cs

```diff
@@ -0,0 +1,158 @@
+
+        /// <summary>
+        /// Initialization handler
+        /// 
+        /// TODO(bengreenier): This could be auto-bound with editor script
+        /// </summary>
+        /// <remarks>
+        /// This is expected to be bound to <see cref="Webrtc.OnInitialized"/> via the UnityEditor
+        /// </remarks>
+        public void OnInitialized()
+        {
+            // cache the peer.
+            this.peer = this.GetComponent<Webrtc>().Peer;
+
+            // setup the video players frame queue references.
+            this.LocalPlayer.FrameQueue = localFrameQueue;
+            this.RemotePlayer.FrameQueue = remoteFrameQueue;
+
+            // setup the peer events.
+            this.peer.LocalI420FrameReady += Peer_LocalI420FrameReady;
+            this.peer.RemoteI420FrameReady += Peer_RemoteI420FrameReady;
+        }
+
+        /// <summary>
+        /// Unity Engine OnDestroy() hook
+        /// </summary>
+        /// <remarks>
+        /// https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnDestroy.html
+        /// </remarks>
+        private void OnDestroy()
+        {
+            // cleanup, if necessary
+            if (this.peer != null)
+            {
+                // cleanup the peer events.
+                this.peer.LocalI420FrameReady -= Peer_LocalI420FrameReady;
+                this.peer.RemoteI420FrameReady -= Peer_RemoteI420FrameReady;
+
+                // cleanup the peer.
+                this.peer = null;
+            }
+        }
+
+        private void Peer_LocalI420FrameReady(IntPtr dataY, IntPtr dataU, IntPtr dataV, IntPtr dataA, int strideY, int strideU, int strideV, int strideA, uint width, uint height)
+        {
+            FramePacket packet = localFrameQueue.GetDataBufferWithoutContents((int)(width * height * 4));
+            if (packet == null)
+            {
+                return;
+            }
+            CopyYuvToBuffer(dataY, dataU, dataV, strideY, strideU, strideV, width, height, packet.Buffer);
+            packet.width = (int)width;
+            packet.height = (int)height;
+            localFrameQueue.Push(packet);
+        }
+
+        private void Peer_RemoteI420FrameReady(IntPtr dataY, IntPtr dataU, IntPtr dataV, IntPtr dataA, int strideY, int strideU, int strideV, int strideA, uint width, uint height)
+        {
+            FramePacket packet = remoteFrameQueue.GetDataBufferWithoutContents((int)(width * height * 4));
+            if (packet == null)
+            {
+                return;
+            }
+            CopyYuvToBuffer(dataY, dataU, dataV, strideY, strideU, strideV, width, height, packet.Buffer);
+            packet.width = (int)width;
+            packet.height = (int)height;
+            remoteFrameQueue.Push(packet);
+        }
+
+        /// <summary>
+        /// Internal yuv data copy routine
+        /// </summary>
+        private void CopyYuvToBuffer(IntPtr dataY, IntPtr dataU, IntPtr dataV,
```

Comment:
>CopyYuvToBuffer [](start = 21, length = 15)

can we gets some comments calling out the specific yuv codec that you are iterating over? or at least the ratio of Y:U:V and how pixel values are laced?